src/System.Numerics.Vectors/tests/Performance/Constructor/GenericVectorConstructorTests.cs

```diff
@@ -0,0 +1,353 @@
+                {
+                    SpanCast<Int16>(spanValues);
+                }
+            }
+        }
+
+        [Benchmark(InnerIterationCount = DefaultInnerIterationsCount)]
+        public static void SpanCastBenchmark_UInt32()
+        {
+            UInt32[] arrValues = GenerateRandomValuesForVector<UInt32>();
+            var spanValues = new ReadOnlySpan<UInt32>(arrValues);
+            foreach (var iteration in Benchmark.Iterations)
+            {
+                using (iteration.StartMeasurement())
+                {
+                    SpanCast<UInt32>(spanValues);
+                }
+            }
+        }
+
+        [Benchmark(InnerIterationCount = DefaultInnerIterationsCount)]
+        public static void SpanCastBenchmark_Int32()
+        {
+            Int32[] arrValues = GenerateRandomValuesForVector<Int32>();
+            var spanValues = new ReadOnlySpan<Int32>(arrValues);
+            foreach (var iteration in Benchmark.Iterations)
+            {
+                using (iteration.StartMeasurement())
+                {
+                    SpanCast<Int32>(spanValues);
+                }
+            }
+        }
+
+        [Benchmark(InnerIterationCount = DefaultInnerIterationsCount)]
+        public static void SpanCastBenchmark_UInt64()
+        {
+            UInt64[] arrValues = GenerateRandomValuesForVector<UInt64>();
+            var spanValues = new ReadOnlySpan<UInt64>(arrValues);
+            foreach (var iteration in Benchmark.Iterations)
+            {
+                using (iteration.StartMeasurement())
+                {
+                    SpanCast<UInt64>(spanValues);
+                }
+            }
+        }
+
+        [Benchmark(InnerIterationCount = DefaultInnerIterationsCount)]
+        public static void SpanCastBenchmark_Int64()
+        {
+            Int64[] arrValues = GenerateRandomValuesForVector<Int64>();
+            var spanValues = new ReadOnlySpan<Int64>(arrValues);
+            foreach (var iteration in Benchmark.Iterations)
+            {
+                using (iteration.StartMeasurement())
+                {
+                    SpanCast<Int64>(spanValues);
+                }
+            }
+        }
+
+        [Benchmark(InnerIterationCount = DefaultInnerIterationsCount)]
+        public static void SpanCastBenchmark_Single()
+        {
+            Single[] arrValues = GenerateRandomValuesForVector<Single>();
+            var spanValues = new ReadOnlySpan<Single>(arrValues);
+            foreach (var iteration in Benchmark.Iterations)
+            {
+                using (iteration.StartMeasurement())
+                {
+                    SpanCast<Single>(spanValues);
+                }
+            }
+        }
+
+        [Benchmark(InnerIterationCount = DefaultInnerIterationsCount)]
+        public static void SpanCastBenchmark_Double()
+        {
+            Double[] arrValues = GenerateRandomValuesForVector<Double>();
```

Comment:
Please use language keywords instead (here and elsewhere): https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/coding-style.md

> We use language keywords instead of BCL types (i.e. `int, string, float` instead of `Int32, String, Single`, etc) for both type references as well as method calls (i.e. `int.Parse` instead of `Int32.Parse`).

Reply:
Same as in case of the coreclr PR that we discussed, this comes from existing code for template generation.

Reply:
In fact, the non-BCL keywords are being emitted by the `type.Name` property used in the template based code generation., so I think we can hardly do anything with that.

Reply:
Example of above is that for the unsupported types test, I have added the following [here](https://github.com/WinCPP/corefx/blob/fff10484b8a85eb64514717f736fe5b472f9677d/src/System.Numerics.Vectors/tests/GenericVectorTests.netcoreapp.tt#L150-L153),
```C#
    Type[] unsupportedTypes = new[]
    {
        typeof(Guid), typeof(DateTime), typeof(char)
    };
```
Please see the above `typeof(char)` of `char`, the language keyword for C# (the language for text template). However, in the places where `<#=type.Name#>` has been used, it resolves to `Char` and not `char` as `Type.Name` emits the BCL type name. Apparently the entire existing tt code that uses looping and complex logic for code generation depends array of `Type` (e.g. supporteTypes array). All this would have to be changed to use some soft of simple text insertion (of language keywords) instead of using reflection while processing the various tt files.